### PR TITLE
Bugfix for issue 8348

### LINKED ISSUE #8348: ColDefs._init_from_array() loses  unsigned info for elements with shape
If an element of an ndarray has shape defined in the dtype, fail to get bzero properly assigned in the returned Column element of ColDefs object.

I believe that the 3 checks should be changed from:
```python
            # Check for unsigned ints.
            bzero = None
            if 'I' in format and ftype == np.dtype('uint16'):
                bzero = np.uint16(2**15)
            elif 'J' in format and ftype == np.dtype('uint32'):
                bzero = np.uint32(2**31)
            elif 'K' in format and ftype == np.dtype('uint64'):
                bzero = np.uint64(2**63)
```
to:
```python
            # Check for unsigned ints.
            bzero = None
            if 'I' in format and ftype.base == np.dtype('uint16'):
                bzero = np.uint16(2**15)
            elif 'J' in format and ftype.base == np.dtype('uint32'):
                bzero = np.uint32(2**31)
            elif 'K' in format and ftype.base == np.dtype('uint64'):
                bzero = np.uint64(2**63)
```

Here is an example which exhibits this behavior:
```python
    foo = np.ndarray((2,), dtype=[('A', '<u4', (2,))])
    col_defs = fits.column.ColDefs(foo)
    assert foo.dtype[0].base == col_defs[0].dtype.base
```

### PR DESCRIPTION
**Pull request by subagonsouth**
Added test coverage for the bug described in issue #8348 and fixed the bug.

EDIT: Fix #8348

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -43,6 +43,9 @@ astropy.io.misc
 astropy.io.fits
 ^^^^^^^^^^^^^^^
 
+- Fixed bug in ``ColDefs._init_from_array() that caused non-scalar unsigned
```
Comment (pllim) [inline CHANGES.rst]: Formatting typo.
```suggestion
- Fixed bug in ``ColDefs._init_from_array()`` that caused non-scalar unsigned
```

Comment (pllim) [reply CHANGES.rst]: Also need to move this entry to a section that is consistent with milestone. Thanks!

**Comment by saimn** [comment]
Thanks @subagonsouth, this looks good, I'm just slightly worried about the use of the undocumented `.base` attribute. It seems (https://stackoverflow.com/questions/52476601/purpose-status-of-the-attribute-numpy-dtype-base) that it should be fine and always give the correct dtype, but maybe we should consider `.subdtype`. @mhvk , as the Numpy expert do you know if `.base` should be avoided ? (and if the stackoverflow interpretation is correct ?)

**Comment by saimn** [comment]
Ok, thanks @mhvk !
Yes, the test could go in `astropy/io/fits/tests/test_table.py`. @subagonsouth - Can you move the test there, and then squash your commits ?

**Comment by subagonsouth** [comment]
I can do the first part, but I am unfamiliar with what squash commits means

**Pull request merged**